File: medium/_1315_ SumofNodeswithEven-ValuedGrandpar.cpp
// 1315. Sum of Nodes with Even-Valued GrandparentSolution

class Solution 
{
public:
	int sumEvenGrandparent(TreeNode* root, int parent_val = 1, int grand_parent_val = 1) 
	{
		int sum = 0;
		if (root)
		{
			if (grand_parent_val % 2 == 0)
			{
				sum += root->val;
			}

			sum += sumEvenGrandparent(root->left, root->val, parent_val);
			sum += sumEvenGrandparent(root->right, root->val, parent_val);
		}			
		return sum;
	}
};

File: medium/_1315_ SumofNodeswithEven-ValuedGrandpar.py
# 1315. Sum of Nodes with Even-Valued Grandparent

"""https://leetcode.com/problems/sum-of-nodes-with-even-valued-grandparent/discuss/477048/JavaC%2B%2BPython-1-Line-Recursive-Solution"""

class Solution:
    def sumEvenGrandparent(self, root: TreeNode, p=1, gp=1) -> int:
            return self.sumEvenGrandparent(root.left, root.val, p)  + self.sumEvenGrandparent(root.right, root.val, p)  + root.val * (1 - gp % 2) if root else 0
